lxd/instances_post.go

```diff
@@ -627,6 +627,21 @@ func createFromBackup(s *state.State, r *http.Request, projectName string, data
 		return response.BadRequest(err)
 	}
 
+	// Check project permissions.
+	err = s.DB.Cluster.Transaction(s.ShutdownCtx, func(ctx context.Context, tx *db.ClusterTx) error {
```

Comment:
I see `r.Context()` more often for the transactions. Any specific reason for picking one over the other? Just curious.

Reply:
In cases where the original request is ended quickly and a background operation is created it is not possible to use r.Context() and so s.ShutdownCtx is the next best thing.

Reply:
A possible future improvement is to be able to get a context.Context from the operation itself and pass it here, so that if the operation is cancelled it cancels this transaction etc.

Reply:
I see, thanks for the clarifications.